src/api/scripts/scripts.api.ts

```diff
@@ -64,6 +64,31 @@ export function fetchScriptVersion({
     });
 }
 
+export async function fetchScriptByNameAndVersionDownload({
+    name,
+    version,
+    expandResponseWith,
+}: IScriptByNameAndVersionPayload & IFetchScriptsOptions) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return get<any>({
+        url: API_URLS.SCRIPT_BY_NAME_VERSION_DOWNLOAD,
+        responseType: 'blob',
+        pathParams: {
+            name,
+            version,
+        },
+        queryParams: toExpandQueryParam(expandResponseWith),
+    }).then((response) => {
+        const url = window.URL.createObjectURL(new Blob([response]));
+        const link = document.createElement('a');
+        link.href = url;
+        // eslint-disable-next-line
+        link.setAttribute('download', 'script_' + name + '_' + version + '.json');
+        document.body.appendChild(link);
+        link.click();
```

Comment:
This looks a bit hacky imo, isn't there a cleaner way of triggering a download? 
@vdwjeff maybe you know this?

Or maybe you can use a package like [file-saver](https://www.npmjs.com/package/file-saver)